src/coreclr/jit/emit.h

```diff
@@ -1534,13 +1534,15 @@ class emitter
             BYTE* idjAddr; // address of jump ins (for patching)
         } idjTemp;
 
-        unsigned idjOffs : 30;    // Before jump emission, this is the byte offset within IG of the jump instruction.
+        unsigned idjOffs : 29;    // Before jump emission, this is the byte offset within IG of the jump instruction.
                                   // After emission, for forward jumps, this is the target offset -- in bytes from the
                                   // beginning of the function -- of the target instruction of the jump, used to
                                   // determine if this jump needs to be patched.
         unsigned idjShort : 1;    // is the jump known to be a short  one?
         unsigned idjKeepLong : 1; // should the jump be kept long? (used for
                                   // hot to cold and cold to hot jumps)
+        unsigned idjIsJmpAlways : 1; // indicates that the jump was added at the end of a BB_ALWAYS basic block
```

Comment:
Is this only used on x64? Then I think, unfortunately, the bits should be ifdef'ed by platform. Of course, the optimization should probably be all-platform.

As I stated before, I'm still uncomfortable with adding a bit in the instrDescs to support a "peephole" optimization. I don't know if there is a better way, but it's not great to add a bit for this very specific purpose.

Reply:
The alternative was to have a separate list. I tried this and was asked to revert the change because it added too much complexity. 
The only other alternative is sniffing every jump by iterating the instructions in the group to make sure that the previous instruction wasn't a call, which has throughput implications.

Reply:
The concern about whether the previous instruction is a `call` is due to the need to preserve x64 ABI requirements regarding unwinding?

Note that avoiding the optimization for any jmp following a call is too conservative: it's legal to remove the jmp-to-next after a call if the next instruction/block isn't in a separate EH region or the epilog. Maybe there's a degenerate case of a sequence of jmp-to-next after a call, so maybe they all disappear? e.g.,
```
   call
   jmp A  // EH region 1
A: jmp B // EH region 1
B: jmp C // EH region 0
C: epilog
```
Here, you can't remove all the jmp-to-next, though one could get converted to a nop:
```
   call
A: nop // EH region 1 -- same EH region as `call`
C: epilog
```